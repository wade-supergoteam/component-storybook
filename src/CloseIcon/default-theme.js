// @flow

export const closeFontSize = '1.5rem';
export const closeFontWeight = 700;
export const closeColor = '#000';
export const closeTextShadow = '0 1px 0 #fff';
