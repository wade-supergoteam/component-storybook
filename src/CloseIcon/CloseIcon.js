// @flow

import React from "react";
import styled from "styled-components";
import themeProp from "../utils/theme";

import { hoverFocus } from "../utils/hover";

import { closeFontSize, closeFontWeight, closeColor, closeTextShadow } from "./default-theme";

export const StyledCloseIcon = styled.button`
  font-size: ${themeProp("closeFontSize", closeFontSize)};
  font-weight: ${themeProp("closeFontWeight", closeFontWeight)};
  line-height: 1;
  color: ${themeProp("closeColor", closeColor)};
  text-shadow: ${themeProp("closeTextShadow", closeTextShadow)};
  opacity: 0.5;

  ${hoverFocus(`
    color: ${themeProp("closeColor", closeColor)};
    text-decoration: none;
    opacity: .75;
  `)};

  /* Opinionated: add "hand" cursor to non-disabled .close elements */
  &:not(:disabled):not(.disabled) {
    cursor: pointer;
  }

  padding: 0;
  background: transparent;
  border: 0;
  -webkit-appearance: none;
`;

const CloseIcon = props => (
  <StyledCloseIcon aria-label="Close" {...props}>
    <span aria-hidden="true">{props.children || "Ã—"}</span>
  </StyledCloseIcon>
);

export default CloseIcon;
