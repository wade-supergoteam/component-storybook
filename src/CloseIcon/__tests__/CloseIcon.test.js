import "jest-styled-components";
import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import CloseIcon from "..";

const customTheme = {
  closeFontSize: "1.75rem",
  closeFontWeight: 700,
  closeColor: "red",
  closeTextShadow: "0 1px 0 blue",
};
test("create default close icon", () => {
  const wrapper = renderer.create(<CloseIcon />);
  expect(wrapper).toMatchSnapshot();
});

test("With custom theme", () => {
  const wrapper = renderer.create(<ThemeProvider theme={customTheme} />);
  expect(wrapper).toMatchSnapshot();
});
