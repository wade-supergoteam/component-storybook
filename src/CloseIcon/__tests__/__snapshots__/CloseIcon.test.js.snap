// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With custom theme 1`] = `null`;

exports[`create default close icon 1`] = `
<button
  aria-label="Close"
  className="sc-bdVaJa uNJFa"
>
  <span
    aria-hidden="true"
  >
    Ã—
  </span>
</button>
`;
